### PR TITLE
804 - account ecocredits

### PR DESCRIPTION
**Pull request by haveanicedavid**
## Description

Closes: regen-network/regen-registry#804

Extracts ecocredits table from regen-network/regen-registry#704 into its own component + reuses in a new route to display ecocredits for any given account

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [x] provided a link to the relevant issue or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please **add
your handle next to the items reviewed if you only reviewed selected items**.*

I have...

- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed code correctness and readability
- [ ] verified React components follow DRY principles 
- [ ] reviewed documentation is accurate
- [ ] reviewed tests
- [ ] manually tested (if applicable)

### CODE REVIEW

File: web-registry/src/components/organisms/EcocreditsTable.tsx [hunk 1]
```diff
@@ -0,0 +1,217 @@
+import React, { useEffect, useState } from 'react';
```
Comment (haveanicedavid) [inline web-registry/src/components/organisms/EcocreditsTable.tsx:1]: most of the code here is copied over from the prior `Ecocredits.tsx` - minus the render prop to display buttons

File: web-registry/src/App.tsx [hunk 2]
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
...
           />
           <Route
-            path="/dashboard/ecocredits"
-            element={<ProtectedRoute component={Ecocredits} />}
+            path="ecocredits/dashboard"
+            element={<ProtectedRoute component={MyEcocredits} />}
+          />
+          <Route
+            path="ecocredits/:accountId"
+            element={<ProtectedRoute component={EcocreditsForAccount} />}
```
Comment (blushi) [inline web-registry/src/App.tsx:111]: this shouldn't be a protected route

File: web-registry/src/App.tsx [hunk 3]
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
             element={<PostPurchase />}
           />
           <Route
-            path="/dashboard/ecocredits"
-            element={<ProtectedRoute component={Ecocredits} />}
+            path="ecocredits/dashboard"
+            element={<ProtectedRoute component={MyEcocredits} />}
```
Comment (blushi) [inline web-registry/src/App.tsx:107]: this one shouldn't be protected either (we only need the keplr wallet to be connected for now, and not the user to be logged in with auth0)

File: web-registry/src/pages/EcocreditsForAccount.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+              ml: 2,
+              display: 'inline',
+              '& a': {
+                fontSize: 18,
+                color: 'black',
+                textDecoration: 'none',
+              },
+            }}
+          >
+            <a
```
Comment (blushi) [inline web-registry/src/pages/EcocreditsForAccount.tsx:68]: I believe the link should also include the `ArrowDownIcon`

**Review by blushi — CHANGES_REQUESTED** [review]

File: web-registry/src/App.tsx [hunk 5]
```diff
@@ -102,8 +103,12 @@ const App: React.FC = (): JSX.Element => {
...
+          />
+          <Route
+            path="ecocredits/:accountId"
```
[the remark this answers is not in the document]
Comment (blushi) [reply web-registry/src/App.tsx:110]: I'm also wondering whether we should add `accounts` sub-route to the path, similarly to most block explorers
```suggestion
            path="ecocredits/accounts/:accountAddress"
```

File: web-registry/src/pages/MyEcocredits.tsx [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+import Section from 'web-components/lib/components/section';
+import { TableActionButtons } from 'web-components/lib/components/buttons/TableActionButtons';
+import { getAccountEcocreditsForBatch, getBatches } from '../lib/ecocredit';
+import { ledgerRestUri } from '../ledger';
+import { EcocreditsTable } from '../components/organisms';
+
+import type { TableCredits } from '../components/organisms';
+
+export const MyEcocredits: React.FC = () => {
+  // const walletContext = useWallet();
```
Comment (blushi) [inline web-registry/src/pages/MyEcocredits.tsx:12]: can we enable this now and get rid of the hard coded wallet?

**Review by blushi — APPROVED** [review]
LGTM

**Review by blushi — CHANGES_REQUESTED** [review]
Just noticing that the current code is actually displaying all the credit batches even if the user has no corresponding balance for them, shouldn't just we display the batches with an actual user balance? cc/ @erikalogie

**Comment by erikalogie** [comment]
Yes, it should be just the ecocredits that the user has a balance for.

**Comment by blushi** [comment]
@erikalogie should we just display a message or something in case the user has no ecocredit at all?

**Comment by erikalogie** [comment]
Yes, it can look like this:
https://www.figma.com/file/seUiQM2SPt78KJiDL0Ea8m/Basic-Ecocredits-Ledger-View?node-id=1467%3A95762

**Review by victordelval — APPROVED** [review]
maybe `EcocreditsByAccount.tsx` instead?

**Comment by haveanicedavid** [comment]
Sorry I missed these change requests @blushi - should now be fixed.

added fallback message: 
image

Also refactored a bit of the fetching logic to DRY up and renamed the component (@victordelval )

**Review by blushi — APPROVED** [review]

**Pull request merged**